### PR TITLE
Adds Mavis support to read trace in JSON form

### PR DESCRIPTION
**Pull request by klingaard**
STF tracing is still not supported yet  This will be the next task/branch.

In this branch, a user can run a `json` file through the simulator.

### CODE REVIEW

**Comment by klingaard** [comment]
closes #7

**Review by knute-sifive — APPROVED** [review]

File: core/Inst.hpp [hunk 1]
```diff
@@ -28,7 +30,13 @@ namespace olympia_core
     class Inst {
     public:
 
-        // The modeler needs to alias a type called "SpartaPairDefinitionType" to the Pair Definition class  of itself
+        // Used by Mavis
+        using InstPtr = sparta::SpartaSharedPointer<Inst>;
+        using PtrType = InstPtr;
```
Comment (aka2en) [inline core/Inst.hpp:35]: I'd like to keep InstPtr declared outside `Inst`. `Inst::InstPtr` looks a bit redundant to use. I'll update it for you

Comment (klingaard) [reply core/Inst.hpp:35]: Sounds good.

**Review by aka2en — APPROVED** [review]

**Comment by klingaard** [comment]
BTW, regression will fail until I get the trace branch merged.  At that point, regression will be back on track.

**Comment by aka2en** [comment]
Yeh I notice that. We could just test JSON tests instead and merge it then.

**Pull request merged**